### PR TITLE
DurationFormatter: Digital Data Provider SourceDataProvider impl

### PR DESCRIPTION
**Pull request by kartva**
Will add bakeddata and testdata in separate PR.

Follow up to #5186.

### CODE REVIEW

File: components/experimental/src/duration/provider/digital.rs [hunk 1]
```diff
@@ -27,10 +27,10 @@ use icu_provider::prelude::*;
 pub struct DigitalDurationDataV1<'data> {
     /// The separator between the hour, minute, and second fields.
     #[cfg_attr(feature = "serde", serde(borrow))]
-    separator: Cow<str, 'data>,
+    pub separator: Cow<str, 'data>,
 
     /// The number of digits to pad the hour field with.
-    hour_padding: u8,
+    pub hour_padding: u8,
```
Comment (younies) [inline components/experimental/src/duration/provider/digital.rs:33]:
```suggestion
    pub leftmost_padding: u8,
    
    pub other_padding: u8,
```

**Comment by kartva** [comment]
Note that `es-CL` has inconsistent hour padding for hm and hms:
```json
"durationUnit-type-hm": {
  "durationUnitPattern": "h:mm"
},
"durationUnit-type-hms": {
  "durationUnitPattern": "hh:mm:ss"
},
```

File: provider/source/data/debug/duration/digital@1/fil.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+  "separator": ":",
+  "hms_hour_padding": 1,
+  "hm_hour_padding": 1,
+  "minute_padding": 1
+}
```
Comment (younies) [inline provider/source/data/debug/duration/digital@1/fil.json:6]: As you can see, all the paddings are equal, so, we can just store one padding and returning an error in case they are different

Comment (younies) [reply provider/source/data/debug/duration/digital@1/fil.json:6]: And as we discussed, if the leftmost padding is different , we can store two paddings in this case.

**Review by younies — CHANGES_REQUESTED** [review]
Good code.

File: provider/source/data/debug/duration/digital@1/en.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  },
+  "hm_padding": {
+    "h": 1,
+    "m": 1
+  },
+  "ms_padding": {
+    "m": 1,
+    "s": 1
+  }
+}
```
Comment (younies) [inline provider/source/data/debug/duration/digital@1/en.json:16]: this can be optimized, but we can do it in another PR: @kartva , just file a but to optimize this.

**Review by younies — APPROVED** [review]

File: provider/source/data/debug/duration/digital@1/en.json (hunk 3 above)
Comment (kartva) [reply provider/source/data/debug/duration/digital@1/en.json:16]: To clarify, you are recommending that I file an issue to optimize this?

Comment (sffc) [reply provider/source/data/debug/duration/digital@1/en.json:16]: I think @younies intended to say "file a bug", so yes. Alternatively, if you already have an issue where you are tracking follow-ups, you can add it there.

Comment (sffc) [reply provider/source/data/debug/duration/digital@1/en.json:16]: This should be easy to bitpack: I think the only possible values are

```rust
pub enum HmsPadding {
    None,
    Hour,
    Minute,
    Second,
    HourMinute,
    HourSecond,
    HourMinuteSecond,
}
```

Are there any locales with patterns such as the following? If not, we can cut down the options further:

- `h:m:ss`
- `h:mm:s`
- `h:m:s`
- `m:s`
